src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -128,54 +146,86 @@ def _validate_child(position, child):
         '''
         return isinstance(child, KernelFunctor)
 
+    def node_str(self, colour=True):
+        '''Construct a text representation of this node, optionally
+        containing colour control codes. Specialise as this node has
+        an additional name argument.
+
+        :param bool colour: whether or not to include colour control \
+            codes. Optional argument that defaults to True.
+
+        :returns: description of this PSyIR node.
+        :rtype: str
+
+        '''
+        return f"{self.coloured_name(colour)}[name=\"{self._name}\"]"
```

Comment:
That's the same as the super implementation, can we delete it?

Reply:
The name used here is different to the parent class. It is mentioned in the docstring.